cmd/main.go

```diff
@@ -1,52 +1,120 @@
 package main
 
 import (
+	"2024_2_ThereWillBeName/internal/pkg/auth"
+	httpHandler "2024_2_ThereWillBeName/internal/pkg/auth/delivery/http"
+	"2024_2_ThereWillBeName/internal/pkg/auth/repo"
+	"2024_2_ThereWillBeName/internal/pkg/auth/usecase"
+	"2024_2_ThereWillBeName/internal/pkg/jwt"
+	"database/sql"
 	"flag"
 	"fmt"
 	"log"
 	"net/http"
 	"os"
 	"time"
+
+	_ "github.com/lib/pq"
+
+	"encoding/hex"
+	"math/rand"
+
+	"github.com/gorilla/mux"
 )
 
 type config struct {
-	port int
-	env  string
+	port    int
+	env     string
+	connStr string
 }
 type application struct {
-	config config
-	logger *log.Logger
+	config      config
+	logger      *log.Logger
+	db          *sql.DB
+	jwtSecret   []byte
+	authUseCase auth.AuthUsecase
 }
 
 func main() {
 	var cfg config
 	flag.IntVar(&cfg.port, "port", 8080, "API server port")
 	flag.StringVar(&cfg.env, "env", "development", "Environment")
+	flag.StringVar(&cfg.connStr, "connStr", "host=localhost port=5433 user=test_user password=1234567890 dbname=testdb_tripadvisor sslmode=disable", "PostgreSQL connection string")
 	flag.Parse()
+
 	logger := log.New(os.Stdout, "", log.Ldate|log.Ltime)
-	app := &application{
-		config: cfg,
-		logger: logger,
+
+	db, err := openDB(cfg.connStr)
+	if err != nil {
+		logger.Fatal(err)
 	}
-	mux := http.NewServeMux()
-	mux.HandleFunc("/healthcheck", app.healthcheckHandler)
+	defer db.Close()
+
+	jwtSecret, err := generateSecretKey(32)
+	if err != nil {
+		logger.Fatal("Error generating secret key:", err)
+	}
+
+	authRepo := repo.NewRepository(db)
+	jwtHandler := jwt.NewJWT(string(jwtSecret))
+	authUseCase := usecase.NewAuthUsecase(authRepo, jwtHandler)
+	h := httpHandler.NewHandler(authUseCase, jwtHandler)
```

Comment:
надо имена типов и соотв конструкторы держать консистентными друг с другом
```suggestion
	authRepo := repo.NewAuthRepository(db)
	jwtHandler := jwt.NewJWT(string(jwtSecret))
	authUseCase := usecase.NewAuthUsecase(authRepo, jwtHandler)
	h := httpHandler.NewAuthHandler(authUseCase, jwtHandler)
```